src/index.js

```diff
@@ -48,10 +51,19 @@ class RadioParser extends EventEmitter {
     if (icyMetaInt) {
       const reader = new StreamReader(icyMetaInt);
 
-      reader.on('metadata', metadata => {
+      reader.on('metadata', (metadata) => {
         this._destroyResponse(response);
         this._queueNextRequest(this.getConfig('metadataInterval'));
-        this.emit('metadata', metadata);
+
+        if (this.getConfig('notifyOnChangeOnly') === true && JSON.stringify(metadata) !== JSON.stringify(this._previousMetadata)) {
+          this.emit('metadata', metadata);
+          this._previousMetadata = metadata;
```

Comment:
Here is the place where you can make JSON.stringify and write it to our property.